### DISCUSSION TITLE
how to use/get a section from selection

### DISCUSSION THREAD

**Discussion by danie-dejager** [junegunn/fzf#3485] [Q&A] [answered]
I have a list of servers with hostname, IP, location and description separated with = (but could be anything)
server1 = us-east1 = web1 http://example.com
server2 = us-east2 = web2 http://example2.com
server3 = us-west1 = sql

How can I make a selection with fzf but have it only use the hostname section to call ssh? All the solutions I've found so far would cut off the other fields and therefore not get displayed.

`ssh server3`

**Comment by LangLangBart** [comment, marked as answer]
> _separated with = (but could be anything)_

- try it with the `delimiter` flag and `FIELD INDEX EXPRESSION`
  - see `man fzf`


| Placeholder | Description |
|  :----: |  :----: |
| `{}` |  current line  |
| `{2}` |  current line with positional field 2  |
| `{2..}` |  current line from the 2nd field to the last field |

| Placeholder expression flags | Description | 
|  :----: |  :----: |
| `+` |  space-separated list of the selected lines |
| `f` | temporary file |
| `s` | preserve the whitespace | 

```bash
cat servers.txt |
  fzf --delimiter '=' --preview 'echo {1};echo {2};echo {3}' --preview-window nohidden
```

<img src="https://github.com/junegunn/fzf/assets/92653266/8c3859ee-9180-4489-abbe-2ce7a98da2fb" width="400">


---

- within `fzf` you can then define a hotkey to perform an `action`, for example `execute` or `become`
  - `fzf's` man page and the the [CHANGELOG.md](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md) will help


```bash
cat servers.txt |
  fzf --delimiter '=' --bind 'enter:become:echo I have selected: {1}'
```

<img src="https://github.com/junegunn/fzf/assets/92653266/4652d455-b3b2-468c-a373-997199ebd846" width="500">

**Reply by danie-dejager** [reply]
Thank you! This works for me.
